server/odc-service/src/main/java/com/oceanbase/odc/service/integration/oauth2/SSOEventHandler.java

```diff
@@ -47,37 +51,43 @@ public boolean support(IntegrationEvent integrationEvent) {
     public void preCreate(IntegrationEvent integrationEvent) {
         Verify.notNull(addableClientRegistrationManager, "addableClientRegistrationManager");
         if (integrationEvent.getCurrentConfig().getEnabled()) {
-            SSOIntegrationConfig decryptConfiguration = getDecryptConfiguration(integrationEvent.getCurrentConfig());
+            IntegrationConfig currentConfig = integrationEvent.getCurrentConfig();
+            SSOIntegrationConfig decryptConfiguration =
+                    getDecryptConfiguration(currentConfig, currentConfig.getEncryption().getSecret());
             addableClientRegistrationManager.addToRegister(decryptConfiguration);
         }
     }
 
     @Override
     public void preDelete(IntegrationEvent integrationEvent) {
         Verify.notNull(addableClientRegistrationManager, "addableClientRegistrationManager");
-        SSOIntegrationConfig decryptConfiguration = getDecryptConfiguration(integrationEvent.getCurrentConfig());
+        SSOIntegrationConfig decryptConfiguration = getDecryptConfiguration(integrationEvent.getCurrentConfig(), null);
         addableClientRegistrationManager.removeRegister(decryptConfiguration.resolveRegistrationId());
     }
 
     @Override
     public void preUpdate(IntegrationEvent integrationEvent) {
         Verify.notNull(addableClientRegistrationManager, "addableClientRegistrationManager");
+        IntegrationConfig preConfig = integrationEvent.getPreConfig();
         IntegrationConfig currentConfig = integrationEvent.getCurrentConfig();
         configurationValidator.checkNotEnabledInDbBeforeSave(currentConfig.getEnabled(),
                 currentConfig.getOrganizationId(), currentConfig.getId());
-        SSOIntegrationConfig ssoIntegrationConfig = getDecryptConfiguration(currentConfig);
+        // current config will not have secret when it is updated, secret can't change, so use preConfig
```

Comment:
`currentConfig` may have `secret` if the secret is modified by user during update operation. So here may need a judgement ?

Reply:
secret is designed not to be updated

Reply:
secret is designed not to be updated